src/jdk.jfr/share/classes/jdk/jfr/events/FileReadIOStatisticsEvent.java

```diff
@@ -0,0 +1,95 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.jfr.events;
+
+import jdk.jfr.Category;
+import jdk.jfr.Description;
+import jdk.jfr.Label;
+import jdk.jfr.Name;
+import jdk.jfr.StackTrace;
+import jdk.jfr.internal.Type;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Name(Type.EVENT_NAME_PREFIX + "FileReadIOStatistics")
+@Label("FileReadIO Statistics")
+@Category({ "Java Application", "Statistics" })
+@Description("Read Rate from the FileInputStream, FileChannelImpl, RandomAccessFile")
+@StackTrace(false)
+public final class FileReadIOStatisticsEvent extends AbstractJDKEvent {
+
+    public static final ThreadLocal<FileReadIOStatisticsEvent> EVENT = new ThreadLocal<>() {
+        @Override
+        protected FileReadIOStatisticsEvent initialValue() {
+            return new FileReadIOStatisticsEvent();
+        }
+    };
+
+    // private variables
+    private static AtomicLong totalReadBytesForPeriod = new AtomicLong(0);
+    private static AtomicLong totalDuration = new AtomicLong(0);
+    private static AtomicLong totalReadBytesForProcess = new AtomicLong(0);
+    
+
+    @Label("Read Rate (Bytes per Sec)")
+    public long readRate;
+
+    @Label("Total Accumulated Read Bytes")
+    public long accRead; // total accumulated
+
+    /* Getters */
+    public static long getTotalReadBytesForProcess() {
+        return totalReadBytesForProcess.get();
+    } 
+
+    public static long getTotalDuration() {
+        return totalDuration.get();
+    }
+
+    public static long getTotalReadBytesForPeriod() {
+        return totalReadBytesForPeriod.get();
+    }
+        
+    public static void setTotalReadBytesForPeriod(long bytesRead, long duration) {
+        totalDuration.addAndGet(duration);
+        totalReadBytesForProcess.addAndGet(bytesRead);
+        totalReadBytesForPeriod.addAndGet(bytesRead);
+    }
+
+    // returning rate
+    public static long getReadRateForPeriod() {
+        // decrement the value with the get the gettotal
+        long result = getTotalReadBytesForPeriod();
+        long interval = getTotalDuration();
+        totalReadBytesForPeriod.addAndGet(-result);
+        if (interval > 0) {
+            totalDuration.addAndGet(-interval);
+            double intervalInSec = (interval * 1.0 / 1_000_000_000);
+            long rRate = (long) (result / intervalInSec);
```

Comment:
I think the cast may not be needed here.

Reply:
@JohnTortugo , The "(result / intervalInSec)" was giving a double and erroring out with "error: incompatible types: possible lossy conversion from double to long". So, I had to explicitly add long to prevent data loss.